src/client/components/pages/externalService.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2022       Ansh Goyal
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import React from 'react';
+import request from 'superagent';
+
+
+const {Alert} = bootstrap;
+class ExternalServices extends React.Component {
+	constructor(props) {
+		super(props);
+		this.alertDetails = props.alertDetails;
+		this.alertType = props.alertType;
+		this.state = {
+			cbPermission: props.cbPermission
+		};
```

Comment:
The alert info should use the component's state rather than being stored as a variable. Then the click handler should update the state accordingly.
```suggestion
		this.state = {
			cbPermission: props.cbPermission,
			alertDetails: props.alertDetails,
			alertType: props.alertType
		};
```

Reply:
Done